ark/phenotyping/som_utils.py

```diff
@@ -464,18 +833,21 @@ def visualize_cluster_data(fovs, channels, base_dir, cluster_dir, cluster_col='c
             Ignored if save_dir is None
     """
 
-    # average the channel values across the cluster column
-    cluster_avgs = compute_cluster_avg(fovs, channels, base_dir, cluster_col, cluster_dir)
+    # average the columns across the cluster column
+    cluster_avgs = compute_cell_cluster_avg(os.path.join(base_dir, cluster_name),
+                                            column_prefix, cell_cluster_col)
 
     # convert cluster column to integer type
-    cluster_avgs[cluster_col] = cluster_avgs[cluster_col].astype(int)
+    cluster_avgs[cell_cluster_col] = cluster_avgs[cell_cluster_col].astype(int)
 
     # sort cluster col in ascending order
-    cluster_avgs = cluster_avgs.sort_values(by=cluster_col)
+    cluster_avgs = cluster_avgs.sort_values(by=cell_cluster_col)
 
     # draw the heatmap
```

Comment:
The more informative heatmap for cells is cells x markers (weighted average - average expression of pixel clusters multiplied by frequency of pixel clusters) - see my code if that's confusing.

Reply:
I'll address this in a separate PR.